pkg/controller/uniteddeployment/allocator.go

```diff
@@ -141,12 +141,13 @@ func getSubsetInfos(nameToSubset *map[string]*Subset, ud *appsv1alpha1.UnitedDep
 // If effective, normalAllocate will be called. It will apply these specified replicas, then average the rest replicas to left unspecified subsets.
 // If not, it will incrementally allocate all of the replicas. The current replicas spread situation will be considered,
 // in order to make the scaling smoothly
-func (s *replicasAllocator) AllocateReplicas(replicas int32, specifiedSubsetReplicas *map[string]int32) (*map[string]int32, bool, string) {
+func (s *replicasAllocator) AllocateReplicas(replicas int32, specifiedSubsetReplicas *map[string]int32) (
+	*map[string]int32, error) {
 	if effective, reason := s.effectiveReplicas(replicas, specifiedSubsetReplicas); !effective {
```

Comment:
How about rename `effectiveReplicas` to `validateReplicas`? And it could return error instead of bool+string.

Reply:
good idea , and i will also change the function comment